### PR TITLE
feat: Added base classes for spring based UI unit tests

### LINKED ISSUE #1337: Create abstract SpringUITest class
Extend the UITest with spring functionalities.

With this addition also the [karibu-testing-v10-spring](https://github.com/mvysny/karibu-testing/tree/master/karibu-testing-v10-spring) mocks should be brought into the project

Acceptance criteria:
- Spring base test class is implemented
- A test using spring atuowiring exists
- documentation is written

### PR DESCRIPTION
**Pull request by mcollovati**
Added JUnit 4 and 5 base test classes that integrates Spring testing
framework and set up a mock Vaadin Spring environment

Fixes #1337

### CODE REVIEW

File: vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/mocks/MockSpringServletService.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ * <ul>
+ * <li>Overrides {@link #isAtmosphereAvailable} to tell Vaadin that we don't
+ * have Atmosphere (otherwise Vaadin will crash)</li>
+ * <li>Provides some dummy value as a root ID via {@link #getMainDivId}
+ * (otherwise the mocked servlet env will crash).</li>
+ * <li>Provides a {@link MockSpringVaadinSession} instead of
+ * {@link com.vaadin.flow.spring.SpringVaadinSession}.</li>
+ * </ul>
+ * The class is intentionally opened, to be extensible in user's library.
+ */
```
Comment (caalador) [inline vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/mocks/MockSpringServletService.java:35]:
```suggestion
 * The class is intentionally opened, to be extensible in user's library.
 *
 * @author mavi
 */
```

File: vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/mocks/MockSpringVaadinSession.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * A Vaadin Session with one important difference:
+ *
+ * <ul>
+ * <li>Creates a new session when this one is closed. This is used to simulate a
+ * logout which closes the session - we need to have a new fresh session to be
+ * able to continue testing. In order to do that, simply override
+ * {@link #close()}, call `super.close()` then call
+ * {@link MockVaadin#afterSessionClose}.</li>
+ * </ul>
+ */
```
Comment (caalador) [inline vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/mocks/MockSpringVaadinSession.java:31]:
```suggestion
 * </ul>
 *
 * @author mavi
 */
```

**Review by caalador — COMMENTED** [review]
Lets add the author for the copied mock classes

File: vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/SpringUIUnit4Test.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ * }
+ * </pre>
+ */
+@RunWith(SpringRunner.class)
+public abstract class SpringUIUnit4Test extends UIUnit4Test {
+
+    @Autowired
+    private ApplicationContext springContext;
+
+    @Before
```
[the remark this answers is not in the document]
Comment (mcollovati) [reply vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/SpringUIUnit4Test.java]: done

File: vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/mocks/MockSpringVaadinSession.java (hunk 2 above)
Comment (mcollovati) [reply vaadin-testbench-unit/src/main/java/com/vaadin/testbench/unit/mocks/MockSpringVaadinSession.java]: done

**Review by caalador — APPROVED** [review]

**Pull request merged**